grails-app/migrations/0.9.x/changelog-2024-10-04-1610-assign-amount-to-invoice-items.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9 http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+
+    <changeSet author="anadolny" id="041020241610-0">
+        <preConditions onFail="MARK_RAN">
+            <and>
+                <columnExists tableName="invoice_item" columnName="amount"/>
+                <not>
+                    <sqlCheck expectedResult="0">
+                        SELECT COUNT(item.id) FROM invoice_item item
+                        WHERE item.amount IS NULL
+                    </sqlCheck>
+                </not>
+            </and>
+        </preConditions>
+
+        <sql>
```

Comment:
can you add a comment here clarifying that this is only allowed because we've already migrated all prepayment invoices and so the only invoice items with a null amount at this point should be non-prepayment ones

Reply:
Also, you can't just do:
```
UPDATE invoice_item
SET amount = quantity * IFNULL(unit_price, 0)
WHERE amount IS NULL;
```